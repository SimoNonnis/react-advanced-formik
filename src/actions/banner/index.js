import { createAction } from 'redux-actions';

export const createSystemBanner = createAction('CREATE_SYSTEM_BANNER');
export const createUserBanner = createAction('CREATE_USER_BANNER');
export const createSuccessBanner = createAction('CREATE_SUCCESS_BANNER');
export const clearBanners = createAction('CLEAR_BANNERS');
