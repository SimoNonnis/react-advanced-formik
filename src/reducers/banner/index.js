import { handleActions } from 'redux-actions';
import { Map } from 'immutable';

import {
  createSystemBanner,
  createUserBanner,
  createSuccessBanner,
  clearBanners,
} from 'actions/banner';

const initialState = Map({
  type: null,
  message: null,
  dismissTime: 7000,
});

const banner = handleActions(
  {
    [clearBanners]: () => initialState,
    [createSystemBanner]: (state, action) => {
      const { payload } = action;
      return state.merge(
        Map({
          type: 'system',
          ...payload,
        })
      );
    },
    [createUserBanner]: (state, action) => {
      const { payload } = action;
      return state.merge(
        Map({
          type: 'user',
          ...payload,
        })
      );
    },
    [createSuccessBanner]: (state, action) => {
      const { payload } = action;
      return state.merge(
        Map({
          type: 'success',
          ...payload,
        })
      );
    },
  },
  initialState
);

export default banner;
export { initialState };
