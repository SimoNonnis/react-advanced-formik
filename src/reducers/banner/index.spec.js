import {
  clearBanners,
  createSystemBanner,
  createUserBanner,
  createSuccessBanner,
} from 'actions/banner';
import banner from './';

describe('Test banner reducer', () => {
  it('returns the initialState when the action is null', () => {
    expect(banner(undefined, {})).toMatchSnapshot();
  });
  it('should set state back to initialState', () => {
    expect(banner(undefined, clearBanners())).toMatchSnapshot();
  });
  it('should set state message to payload and type to system', () => {
    expect(banner(undefined, createSystemBanner({ message: 'Error message' }))).toMatchSnapshot();
  });
  it('should set state message to payload and type to user', () => {
    expect(banner(undefined, createUserBanner({ message: 'Error message' }))).toMatchSnapshot();
  });
  it('should set state message to payload and type to success', () => {
    expect(banner(undefined, createSuccessBanner({ message: 'Error message' }))).toMatchSnapshot();
  });
});
