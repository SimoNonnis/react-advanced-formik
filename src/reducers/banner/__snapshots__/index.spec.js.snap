// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test banner reducer returns the initialState when the action is null 1`] = `
Object {
  "dismissTime": 7000,
  "message": null,
  "type": null,
}
`;

exports[`Test banner reducer should set state back to initialState 1`] = `
Object {
  "dismissTime": 7000,
  "message": null,
  "type": null,
}
`;

exports[`Test banner reducer should set state message to payload and type to success 1`] = `
Object {
  "dismissTime": 7000,
  "message": "Error message",
  "type": "success",
}
`;

exports[`Test banner reducer should set state message to payload and type to system 1`] = `
Object {
  "dismissTime": 7000,
  "message": "Error message",
  "type": "system",
}
`;

exports[`Test banner reducer should set state message to payload and type to user 1`] = `
Object {
  "dismissTime": 7000,
  "message": "Error message",
  "type": "user",
}
`;
